### PR TITLE
Osm custom params

### LINKED ISSUE #271: Add legacy XYZ layer for backward compatibility
for backward compatibility it would be nice to add legacy xyz layer
See pull request  #270

### PR DESCRIPTION
**Pull request by didier-coignet-dict**
* adds ability to use custom params with osm layer
* convert old api xyz layer to osm 
fixes #271

### CODE REVIEW

File: examples/src/test/resources/examples/osm_custom_params/oldApi-requestData.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{
+  "app" : "mapbox xyz old api",
+  "srs": "EPSG:3857",
+  "layout": "A4 landscape",
+  "outputFilename": "exemplemapbox",
+  "dpi": 72,
+  "outputFormat": "pdf",
+  "layers": [
+    {
+      "baseURL": "https://api.tiles.mapbox.com/v4/tmcw.map-7s15q36b/",
```
Comment (jesseeichar) [inline examples/src/test/resources/examples/osm_custom_params/oldApi-requestData.json]: I don't care for tests that require external resources.  For example the server could crash or be removed.  Could you do like I did for OSM?  I downloaded the tiles that are required for the example and put them in:  `examples/src/main/webapp/data/www/osm` 

You can do the same and put them in   `examples/src/main/webapp/data/www/mapbox` then change baseURL to be: `"http://localhost:9876/e2egeoserver/www/mapbox"`

See the OSM example: `examples/src/test/resources/examples/print_osm_new_york_EPSG_900913/requestData-scale-osm.json`

Comment (jesseeichar) [inline examples/src/test/resources/examples/osm_custom_params/oldApi-requestData.json]: I know it doesn't perfectly test mapbox but in the long run it is more reliable I think

**Comment by jesseeichar** [comment]
Looks good to me.  Just a minor comment

File: examples/src/test/resources/examples/osm_custom_params/oldApi-requestData.json (hunk 1 above)
Comment (didier-coignet-dict) [inline examples/src/test/resources/examples/osm_custom_params/oldApi-requestData.json]: I agree its better an independant test

**Comment by jesseeichar** [comment]
I will merge as soon as the test does not depend on an external server.

**Comment by jesseeichar** [comment]
See #314

**Pull request closed without merging**